recipes/cudatoolkit-dev/cudatoolkit-dev-post-install.py

```diff
@@ -0,0 +1,332 @@
+""" BSD 2-Clause License
+
+Copyright (c) 2018 Onwards, Quansight, LLC
+Copyright (c) 2017, Continuum Analytics, Inc.
+
+All rights reserved.
+
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. """
+
+from __future__ import print_function
+import os
+import sys
+import shutil
+import urllib.parse as urlparse
+from pathlib import Path
+import subprocess
+from conda.exports import download, hashsum_file
+import stat
+
+
+def set_chmod(file_name):
+    # Do a simple chmod +x for a file within python
+    st = os.stat(file_name)
+    os.chmod(file_name, st.st_mode | stat.S_IXOTH)
+
+
+def create_dir(new_dir):
+    try:
+        os.makedirs(new_dir)
+    except FileExistsError:
+        pass
+
+
+def copy_files(src, dst):
+    try:
+        if (os.path.isfile(src)):
+            set_chmod(src)
+            shutil.copy(src, dst)
+    except FileExistsError:
+        pass
+
+
+config = {}
+versions = ["9.2"]
```

Comment:
These version numbers should all be filled in with the values based on what the version is in the meta.yaml.  That is why I think this script should either read in `$PKG_VERSION` from the environment (if it is available) or this should be a template file (probably in Jinja2) that then is rendered with the correct version numbers by a build.sh script.

Reply:
This is fairly easy to fix. I will update it to use the `$PKG_VERSION`.

Reply:
Actually, I am going to look into refactoring some part of cudatoolkit-dev-post-install.. I adapted it from the build script from numba conda recipe and their script (https://github.com/numba/conda-recipe-cudatoolkit/blob/master/scripts/build.py#L18) uses an interesting method to update the recipe as the new versions of CUDA are released. I believe this method ends up forcing them to hard code some variables such as `versions`.